### PR TITLE
refactor: edit squad page

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Refactored the account danger zone by using a newly introduced reusable component.
- Refactored the Squad page header components to make them reusable.
- Apply edit Squad changes.

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

MI-493

### Preview domain
https://mi-498.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/hooks/useDeleteSquad.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ interface UseDeleteSquadModal {
 }
 
 type UseDeleteSquadProps = {
-  squad: Squad;
+  squad: Partial<Squad>;
```
Comment (sshanzel) [inline packages/shared/src/hooks/useDeleteSquad.ts:15]: Bear with me, I will introduce a PR to refactor the Squad form as it is quite difficult to manage/maintain.

**Review by nensidosari — APPROVED** [review]
Looks good!

File: packages/shared/src/components/profile/AccountDangerZone.tsx [hunk 2]
```diff
@@ -1,64 +1,45 @@
...
 
+const Important = () => (
+  <>
+    Important: deleting your account is unrecoverable and cannot be undone. Feel
+    free to contact{' '}
+    <a
+      className="text-text-link"
+      href="mailto:support@daily.dev?subject=I have a question about deleting my account"
+      target="_blank"
+      rel="noopener"
```
Comment (rebelchris) [inline packages/shared/src/components/profile/AccountDangerZone.tsx:18]:
```suggestion
      rel={anchorDefaultRel}
```

File: packages/shared/src/components/squads/settings/SquadDangerZone.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+const Important = () => (
+  <>
+    Important: deleting your Squad is unrecoverable and cannot be undone. Feel
+    free to contact{' '}
+    <a
+      className="text-text-link"
+      href="mailto:support@daily.dev?subject=I have a question about deleting my Squad"
+      target="_blank"
+      rel="noopener"
```
Comment (rebelchris) [inline packages/shared/src/components/squads/settings/SquadDangerZone.tsx:20]:
```suggestion
      rel={anchorDefaultRel}
```

**Review by rebelchris — APPROVED** [review]
Nice great work on splitting out some comments.
Approving, but would be great if you can add the rel={anchorDefaultRel}

File: packages/shared/src/components/profile/AccountDangerZone.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/profile/AccountDangerZone.tsx:18]: Oh, cool. Just copy-pasted this block and didn't realize it. Yep much better to use the common one.

**Pull request merged**